metrics/config_observability.go

```diff
@@ -83,6 +87,16 @@ func NewObservabilityConfigFromConfigMap(configMap *corev1.ConfigMap) (*Observab
 		return nil, err
 	}
 
+	if raw, ok := configMap.Data["logging.enable-request-log"]; ok {
+		if strings.EqualFold(raw, "true") && oc.RequestLogTemplate != "" {
+			oc.EnableRequestLog = true
+		}
+	} else {
+		if oc.RequestLogTemplate != "" {
+			oc.EnableRequestLog = true
+		}
+	}
```

Comment:
```suggestion
	} else if oc.RequestLogTemplate != "" {
		// If logging.enable-request-log is not explicitly set, defaults to true if the template is not empty.
		oc.EnableRequestLog = true
	}
```

Reply:
I didnt like to put the if there because usually I would expect an `else` somewhere and it helps readability imho, but I can optimize it.